toDoApp/app/src/main/java/hieunguyen/com/todoapp/MainActivity.java

```diff
@@ -0,0 +1,181 @@
+package hieunguyen.com.todoapp;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.TextView;
+
+public class MainActivity extends AppCompatActivity {
+
+    MyAdapter itemsAdapter;
+    ListView listView;
+    MainService dbService = new MainService(this);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        itemsAdapter = new MyAdapter(this, dbService);
+
+        updateListHeader();
+
+        listView = (ListView) findViewById(R.id.lvItems);
+        listView.setAdapter(itemsAdapter);
+
+        // On tap, allow user to edit task
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
```

Comment:
Your activity would look cleaner if you define each one of these classes in a separate file. Like we do here
https://github.com/OlinMobileProto/Lab1/blob/master/app/src/main/java/com/example/cwallace/grocerylist/clickListeners/DeleteEntryListener.java